### ISSUE TITLE
Accessibility Issue: "Report form general"

### ISSUE THREAD

**Issue by larrykind** [globaleaks/GlobaLeaks#3374]
### What version of GlobaLeaks are you using?

4.10.18 - debelopment environment

### What browser(s) are you seeing the problem on?

All

### What operating system(s) are you seeing the problem on?

Windows, macOS, Android, iOS, Linux

### Describe the issue

Hello, 
this accessibility report is about accessibility issue on the report form element, we tried with various form configurations.
Please mark it as "Accessibility" issue.

| Index | Issue Description | Ref. Check UNI CEI EN 301549:2021 | conformity level |
| ---- | ---- | ---- | ----  |
| 1 | For the most part of the input fields the element <label>, if present, isn't associated to its relative field trhough the "for" attribute. ---> Create everywhere this association. The "for" must be associated to the ID attribute of the relative input field. | C.9.3.3.2 | A |
| 2 | Form: multiple field -  Button "+" has not an accessible name: --> add aria-label="add" followed by an explication text to be avaluated,  for example  "add a new section "[name of multiple section].| C.9.4.1.2 | A |
| 3 | When buttons "next" or "previous"  at the end of the page, the focus is not sent to the beginning of the page, but it remains at the end of the page --> focus shoud be winded to the page beginning. | C.9.2.4.3 | A |
| 4 | Form : on every steps elements are not grouped logically: --> use fieldset and legend to describe semantical group fields. For example: everything that is "added" with the button "add" should be marked as "fieldset".| C.9.1.3.1 | A |
| 5 | In the input fields attributes WAI-ARIA tells the screen readers that inserted data is not valid even if the field is yet empty.| C.9.4.1.2 | A |
| 6 | ID of elements are not unique| C.9.1.4.11 | AA |
| 7 | Mandatory fields are marked as "mandatory" in a non correct way set "required" attribute on mandatory fields.| C.9.1.3.1 | A |
| 8 | The questionnaire filling error system is managed in a non correct way (visually only), entirely with no semantics. ---> Need to implement an accessible methodology, for exemple using role="alert" to describe text message generated at the page beginning, and the attribute "aria-describedby" to link the input field to the relative error message. Ex: https://www.w3.org/WAI/WCAG21/working-examples/aria-invalid-data-format/ | C.9.3.3.1 | A |
| 9 | Questionnaire: in grouped questions, errors on internal questions are not marked, just error on the group question is fired | C.9.3.3.1 | A |
| 10 | *Attachment object: this object is realised with "input type file" but label is not referenced with the "for" attribute to the input field. Object is to be verified after the fix because it's not working with some browsers | C.9.3.3.2 | A |
| 11 | The "X" to delete a loaded attachment doesnt have an accessible name: ---> add  aria-label="delete attachment". | C.9.1.1.1 | A |
| 12 | the functionality "copy" of the 16-digit code is a button that does not communicate its funcionality. The copy works but users have no feedback. The code should be written in a separate text so users can use also the native functionality "copy/paste". Eventually provide this functionality with separate button.| C.9.4.1.2 | A |
| 13 | The loading functionality component (ex: attachment loagding) has no semantics. It should be managed with an element who owns the attribute aria-live="polite". It should be empty at page loading and then valued with a string like "please wait" when the layer with the loading symbol is shown.| C.9.4.1.2 | A |

### Proposed solution

Solutions are proposed in the upper table, the points with an asterisk are those on which we think we can act independently.

**Comment by nicolaisweb** [comment]
the issue n.6: "ID of elements are not unique" is not fixed yet in release 4.12.8

**Comment by evilaliv3** [comment]
@nicolaisweb: would you please more detail?

**Comment by nicolaisweb** [comment]
> @nicolaisweb: would you please more detail?

In the whislteblower report review page, there are 3 IDs with id `#TipInfoBox`, please tell me if you need more details.
